platform/admin/core/admin-core-migration-commands/src/main/java/org/codice/ddf/migration/commands/MigrationCommand.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration.commands;
+
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.codice.ddf.configuration.migration.ConfigurationMigrationService;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationMessage;
+import org.codice.ddf.migration.MigrationSuccessfulInformation;
+import org.codice.ddf.migration.MigrationWarning;
+import org.codice.ddf.security.common.Security;
+import org.fusesource.jansi.Ansi;
+import org.fusesource.jansi.Ansi.Attribute;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Provides common methods and instance variables that migration commands can use.
+ */
+public abstract class MigrationCommand implements Action {
```

Comment:
This class and the ones that extend it appear to duplicate a lot of the security functionality that exists in the SubjectCommands class in catalog. Maybe we should consider moving that out of catalog so that commands in other areas of the system can have access to the same stuff.

Reply:
This is the original code for the export command class which got abstracted out. It is pretty much empty as it is. I don't see what would be gained.